### PR TITLE
feat: add bearer reader

### PR DESCRIPTION
**Pull request by cfabianski**
Add Bearer Reader

https://github.com/bearer/bearer

Requires this one to be released https://github.com/Bearer/bearer/pull/1401

### CODE REVIEW

**Review by darkspirit510 — CHANGES_REQUESTED** [review]

File: plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/BearerReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * href="https://owasp.org/www-project-benchmark/">https://owasp.org/www-project-benchmark/</a>.
+ *
+ * <p>The OWASP Benchmark is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation, version 2.
+ *
+ * <p>The OWASP Benchmark is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE. See the GNU General Public License for more details.
+ *
+ * @author Sascha Knoop
```
[the remark this answers is not in the document]
Comment (cfabianski) [reply plugin/src/test/java/org/owasp/benchmarkutils/score/parsers/BearerReaderTest.java]: Thanks @darkspirit510 
It's fixed now :)

File: plugin/src/main/java/org/owasp/benchmarkutils/score/parsers/BearerReader.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        switch (cwe) {
+            case 327:
+                return CweNumber.WEAK_HASH_ALGO;
+            default:
+                return cwe;
+        }
+    }
+
+    private TestCaseResult parseBearerFindings(JSONObject result) {
+        /*
```
[the remark this answers is not in the document]
Comment (cfabianski) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/parsers/BearerReader.java]: Thanks @darkspirit510 
Yeah I removed it. Test case is more than enough :)

**Comment by davewichers** [comment]
@darkspirit510 - You OK with this MR now so I can merge it?

**Comment by darkspirit510** [comment]
@davewichers the code looks good, but as you wrote in https://github.com/OWASP-Benchmark/BenchmarkJava/pull/214, I did not yet manage to create/test with a real file.

**Comment by darkspirit510** [comment]
@davewichers I just build the code from Bearer's repo and was able to scan the file.

**Review by darkspirit510 — APPROVED** [review]

**Comment by cfabianski** [comment]
> @davewichers I just build the code from Bearer's repo and was able to scan the file.

Hey @darkspirit510 thanks for this and sorry for the lack of response :(
Yeah the release went probably after you tried https://github.com/Bearer/bearer/releases/tag/v1.31.0

**Pull request merged**